src/query/query_eval.rs

```diff
@@ -31,74 +31,93 @@ fn eval_latest<'a>(
-        })
+        .filter(|packet| lookup_filter(packet, &test, &value,&lookup_field))
         .collect())
 }
 
+fn lookup_filter(packet: &Packet, test: &Test, value: &Literal, lookup: &Lookup) -> bool {
+    match (test, lookup) {
+        (Test::Equal, Lookup::Id) => match value {
+            Literal::String(str) => packet.id == *str,
+            _ => false
+        },
+        (Test::NotEqual, Lookup::Id) => match value {
+            Literal::String(str) => packet.id != *str,
+            _ => false
+        },
+        (_, Lookup::Id) => false,
+        (Test::Equal, Lookup::Name) => match value {
+            Literal::String(str) => packet.name == *str,
+            _ => false
+        },
+        (Test::NotEqual, Lookup::Name) => match value {
+            Literal::String(str) => packet.name != *str,
+            _ => false
+        },
+        (_, Lookup::Name) => false,
+        (test, Lookup::Parameter(param_name)) => packet.test_parameter(param_name, test, value),
+    }
+}
+
 impl Packet {
-    /// Check if a packet parameter is equal to a test value
+    pub fn get_parameter(&self, param_name: &str) ->  Option<Literal> {
+        if let Some(params) = &self.parameters {
+            match params.get(param_name)? {
+                JsonValue::Number(number) => {
+                    Some(Literal::Number(number.as_f64()?))
+                },
+                JsonValue::Bool(bool) => {
+                    Some(Literal::Bool(*bool))
+                },
+                JsonValue::String(string) => {
+                    Some(Literal::String(string))
+                },
+                _ => None // Parameters must be number, bool or string
+            }
+        } else {
+            None
+        }
+    }
+
+    /// Run a comparison test on a packet parameter and a test value
     ///
-    /// This will get the parameter `param_name` from the current Packet and then test for
-    /// equality of the test value `value`.
+    /// This will get the parameter `param_name` from the current Packet and then run the
+    /// specified comparison `test` with the test value `value`.
     ///
     /// # Arguments
     /// * `param_name` - A string slice that holds the name of the parameter to test
-    /// * `value` - A LookupRhs which holds the value to check equality for, can be a boolean, a
+    /// * `test` - The type of test to run, ==, !=, <, <=, > or >=
+    /// * `value` - A Literal which holds the value to check equality for, can be a boolean, a
     ///             string or a number
     ///
     /// # Return
     /// * bool - true if the current packet has a parameter called `param_name` and its value
-    ///          is equal to the LookupRhs.
-    fn parameter_equals(&self, param_name: &str, value: &LookupRhs) -> bool {
+    ///          passes the specified test with the input test `value`.
+    fn test_parameter(&self, param_name: &str, test: &Test, value: &Literal) -> bool {
```

Comment:
This doesn't allow you to do < or > on strings to get inequality matches alphabetical order? Do we specifically want to not support that? I suppose there's not a obvious use case for it!

Reply:
Yep that's right, have actively stopped that in the R impl too. Decided to turn it off just to avoid potential confusion from ambiguity of ordering. Is `"A" < "a"`? `"123" < "abc"`? `".." < "a"`? And are they consistent over langauges? In particular in Rust

```
fn main() {
    println!("{}", "a" < "A");
}
```
Gives `false` but in R

```
"a" < "A"
```
prints `TRUE` :fearful: 

So have deliberately not supported this for now but we could add it in future, but I am not sure how often people actually do this kind of comparison

Reply:
This is actually locale dependent

```r
x <- c("Bernard", "bernard", "béatrice", "Béatrice")
withr::with_locale(c(LC_COLLATE = "fr-FR"), sort(x))
withr::with_locale(c(LC_COLLATE = "C"), sort(x))
```

so not something we should encourage as it will differ based on people's machines!